packages/engine/src/signature/signature.cairo

```diff
@@ -441,9 +468,11 @@ pub fn parse_signature(sig_bytes: @ByteArray) -> Result<Signature, felt252> {
     if s_sig == 0 {
         return Result::Err('invalid sig: S is zero');
     }
-    if sig_len != r_len + s_len + 6 {
+
+    if sig_bytes.len() != sig_bytes[3].into() + sig_bytes[sig_bytes[3].into() + 5].into() + 6 {
```

Comment:
Would it be more efficient to do : `if sig_bytes.len() != r_len + s_len + 6 {`

Reply:
Because we are mutating `r_len` and `s_len` when stripping zeros, so  I use the original `r_len` and `s_len` to match the btcd implementation